### ISSUE TITLE
[css-backgrounds] a `no-repeat` option for `border-image-repeat`

### ISSUE THREAD

**Issue by jsnkuhn** [w3c/csswg-drafts#7457]
https://drafts.csswg.org/css-backgrounds-3/#border-image-repeat

a `no-repeat` option for `border-image-repeat` that would center the sliced edges like `repeat` does but would NOT fill in the empty space with repeated tiles.

this would allow developers to create border-images with centered edges like this:

![Screenshot 2022-07-04 at 20-44-38 Dialogue - Hollow Knight Interface In Game](https://user-images.githubusercontent.com/1286791/177408953-2c96c51c-906c-4b57-b9cc-b0c503076f5b.png)

![Screenshot 2022-07-05 at 15-17-08 hollow_knight_charm_equip_screen webp (WEBP Image 920 × 518 pixels)](https://user-images.githubusercontent.com/1286791/177409139-cb37ee6d-f356-43c2-a8fe-9679645f0b48.png)

**Comment by fantasai** [comment]
What happens if the box is too small to accommodate the entire center image? Do we just clip it, or do we want to scale it down?

**Comment by jsnkuhn** [comment]
ahh, honestly hadn't thought about the `fill`. I assume if a designer went to the trouble to make a border image they probably wouldn't want the browser to just arbitrarily cut it off so "scale it down" sounds like the better option here.

**Comment by jsnkuhn** [comment]
It looks like there are 3 different possibilities right now for when there isn't room for a whole iteration of the edge image. `stretch` and `round` scale it down, `repeat` clips from each size and `space` just removes the image altogether. 

The current `space` behavior (just removing it) is probably most desired for `no-repeat`.

![GIF](https://user-images.githubusercontent.com/1286791/178069035-11ff715f-7d7a-4632-9ad6-0eab052894bf.gif)

**Comment by jsnkuhn** [comment]
some jury-rigged live `no-repeat` examples:

https://codepen.io/jsnkuhn/pen/ExEPBzX?editors=0110

**Comment by jsnkuhn** [comment]
Not entirely sure whether this would be a use case here or not. Looking at the centered thing on the top border and wondering if that's really part of the border or if it's a separate element.

![RFShrJQ](https://user-images.githubusercontent.com/1286791/183132093-0b5cb8b1-61d9-451d-a4f1-43a40ad9d2d9.png)

**Comment by SebastianZ** [comment]
> The current `space` behavior (just removing it) is probably most desired for `no-repeat`.

I agree with that. Though I can imagine that there may be cases in which scaling it down or clipping it could be desired results.

Having said that, to let this feature find its way into the spec., I think it'd be best to add it with the removing behavior first.
If use cases for the other behaviors rise, we can still add them later.

Sebastian

**Comment by SebastianZ** [comment]
@jsnkuhn The example in your last comment _could_ be covered by having clipping behavior for `no-repeat` Though that would have the downside that the text is part of the image. And as an author you'd probably want to have multiple such elements with different titles in them.
Also, you'd have to create a rather wide image to avoid any gaps between the corner images and the centered one.

For cases like that (but maybe without title), it would probably better to allow multiple border images. So you could define the first one using `stretch` and the second one with `no-repeat`.

Sebastian

**Comment by jsnkuhn** [comment]
Ran into 2 other examples with centered design elements but once again not sure if `no-repeat` is what's needed for these. It almost seems to me like in these cases what's needed is a 5x5 sliced image instead of a 3x3. 

I.E. slices for fixed corners and the center sizes but allow the remaining edge bits to stretch to fill the space?

![Screenshot 2023-01-02 at 17-06-58 Disney Dreamlight Valley Buy Today](https://user-images.githubusercontent.com/1286791/210287716-a2e15f46-e9d0-4c04-b425-c68766267045.png)

![Screenshot 2023-01-02 at 16-39-24 Lost Ark - Free to Play MMO Action RPG](https://user-images.githubusercontent.com/1286791/210287720-dc8021b9-9422-4296-84e7-1a7881900d0e.png)

**Comment by jsnkuhn** [comment]
https://ffvii-remake-intergrade.square-enix-games.com/en-us/

![image](https://user-images.githubusercontent.com/1286791/227198893-e7a03776-28e5-4724-95c4-4d5fa670363b.png)

Currently done with a bunch of presentational divs. This one seems even more like it would really need the 5x5 slice. or 3x3 for just the corners and the solid line plus a single `::after` for the centered diamond.

**Comment by jsnkuhn** [comment]
https://www.zelda.com/tears-of-the-kingdom/

This might be more of a `mask-border` situation but it still has the centered style element. 

![image](https://user-images.githubusercontent.com/1286791/231836297-0a57e1d7-b854-41f8-8814-3f82114fedd1.png)

**Comment by SebastianZ** [comment]
@jsnkuhn Your two examples before the last one seem to be better served through a 5x5 grid or multiple border images instead of a `no-repeat` keyword. So maybe we should split that out into a separate issue.

Sebastian

**Comment by jsnkuhn** [comment]
it occurs to me that `no-repeat` probably isn't specific enough of a keyword. The non repeated image fragment could technically be left aligned, right aligned or centered in the top edge. `no-repeat-center-aligned` seems like a bit much... And I don't think I've ever seen a case where something like `no-repeat-left-align` or `no-repeat-right-align` would be used.  So... maybe `center` is a better keyword here?

It also seems worth pointing out that this should apply to both `border-image-repeat` and `mask-border-repeat`
